Exec/radiation_tests/Rad2Tshock/Prob_3d.f90

```diff
@@ -79,38 +79,73 @@ subroutine ca_initdata(level,time,lo,hi,nscal, &
   real(rt)         :: xlo(3), xhi(3)
   
   integer :: i,j,k
-  real(rt)         :: xcell, rhoInv
+  real(rt)         :: length_cell,length_min, rhoInv
   type(eos_t) :: eos_state
-
+    
+  
   do k = lo(3), hi(3)  
      do j = lo(2), hi(2)
         do i = lo(1), hi(1)
-  
-           xcell = xmin  + delta(1) * (dble(i) + 0.5e0_rt)
+
+           !Provides the simulation to be run in the x,y,or z direction
+           !where length direction is the length side in a square prism  
+           if (idir == 1) then
+              length_cell = xmin + delta(1) * (dble(i) + 0.5e0_rt)
+           else if (idir == 2) then
+              length_cell = ymin + delta(2) * (dble(j) + 0.5e0_rt)
+           else if (idir == 3) then
+              length_cell = zmin + delta(3) * (dble(k) + 0.5e0_rt)
+           else
+              stop "Invalid Direction, input integer int_dir = [1,3] ."
```

Comment:
Please use amrex_error("Invalid direction...") rather than the Fortran stop statement.